src/server/test/db/cikVaryTests.js

```diff
@@ -0,0 +1,150 @@
+const { redoCikVary } = require('../../services/graph/redoCik');
+const Conversion = require('../../models/Conversion');
+const Unit = require('../../models/Unit');
+const ConversionSegment = require('../../models/ConversionSegment');
+
+// Insert test units for IDs 10, 1, 2
+async function insertTestUnits(conn) {
+	await new Unit(undefined, 'Unit 10', 'Unit 10', Unit.unitRepresentType.QUANTITY, 1000, Unit.unitType.METER, 'Suffix 10', Unit.displayableType.ADMIN, true, 'Note 10').insert(conn);
+	await new Unit(undefined, 'Unit 1', 'Unit 1', Unit.unitRepresentType.QUANTITY, 1001, Unit.unitType.UNIT, 'Suffix 1', Unit.displayableType.ADMIN, true, 'Note 1').insert(conn);
+	await new Unit(undefined, 'Unit 2', 'Unit 2', Unit.unitRepresentType.QUANTITY, 1002, Unit.unitType.UNIT, 'Suffix 2', Unit.displayableType.ADMIN, true, 'Note 2').insert(conn);
+}
+
+// Insert multiple segments for 10->1 and 1->2 into conversion_segments
+async function insertTestSegments(conn) {
+	const unit10Id = (await Unit.getByName('Unit 10', conn)).id;
+	const unit1Id = (await Unit.getByName('Unit 1', conn)).id;
+	const unit2Id = (await Unit.getByName('Unit 2', conn)).id;
+
+	// For 10 -> 1
+
+	// Split at 2020-01-01 00:00:00
+	await ConversionSegment.splitLater(
+		unit10Id, unit1Id, null, 2, 0, null,
+		'-infinity', 'infinity', '2020-01-01 00:00:00', conn
+	);
+
+	// Split at 2020-06-01 00:00:00 (on the segment starting at 2020-01-01)
+	await ConversionSegment.splitLater(
+		unit10Id, unit1Id, null, 3, 0, null,
+		'2020-01-01 00:00:00', 'infinity', '2020-06-01 00:00:00', conn
+	);
+
+	// Split at 2021-01-01 00:00:00 (on the segment starting at 2020-06-01)
+	//await ConversionSegment.splitLater(
+	//	unit10Id, unit1Id, null, 4, 0, null,
+	//	'2020-06-01 00:00:00', 'infinity', '2021-01-01 00:00:00', conn
+	//);
+
+	await ConversionSegment.splitLater(
+		unit1Id, unit2Id, null, 6, 0, null,
+		'-infinity', 'infinity', '2020-01-01 00:00:00', conn
+	);
+
+	await ConversionSegment.splitLater(
+		unit1Id, unit2Id, null, 7, 0, null,
+		'2020-01-01 00:00:00', 'infinity', '2020-06-01 00:00:00', conn
+	);
+
+	await ConversionSegment.splitLater(
+		unit1Id, unit2Id, null, 8, 0, null,
+		'2020-06-01 00:00:00', 'infinity', '2021-01-01 00:00:00', conn
+	);
+}
+
+// Insert test conversions using the Conversion class
+async function insertTestConversions(conn) {
+	const unit10Id = (await Unit.getByName('Unit 10', conn)).id;
+	const unit1Id = (await Unit.getByName('Unit 1', conn)).id;
+	const unit2Id = (await Unit.getByName('Unit 2', conn)).id;
+	const conversionPreInsert1 = new Conversion(unit10Id, unit1Id, false, 'note');
+	const conversionPreInsert2 = new Conversion(unit1Id, unit2Id, false, 'note');
+	await Promise.all([
+		conversionPreInsert1.insert(null, 1, 0, "", conn),
+		conversionPreInsert2.insert(null, 1, 0, "notes", conn)
+	]);
+}
+
+mocha.describe('CIK Vary Chaining', () => {
+
+	mocha.describe('Insert test unit', function () {
```

Comment:
I'm not sure the second describe is needed. I tried commenting it out and it worked fine with a good description. Applies to other tests too.